src/neo/Network/P2P/TaskSession.cs

```diff
@@ -17,14 +17,16 @@ internal class TaskSession
         public bool HasTask => Tasks.Count > 0;
         public uint StartHeight { get; }
         public uint LastBlockIndex { get; set; }
+        public bool ShouldAskForMemPool { get; set; }
```

Comment:
Once it is set to `false`, it will not be used again. So it is a one-time variable. I think we should add a read-only flag `IsFullNode`, and then determine in the `TaskManager` whether to ask for the memory pool.